File: WildFire.Web/src/app/DTOs/UserWeatherRequest.ts
import { LightningEvent } from "../models/LightningEvent";
import { RainEvent } from "../models/RainEvent";
import { WindEvent } from "../models/WindEvent";

export interface UserWeatherRequest{
    LightningEvent: LightningEvent;
    RandomLightning: boolean;
    WindEvent: WindEvent;
    RandomWind: boolean;
    RainEvent: RainEvent;
    RandomRain: boolean;
}

File: WildfireSimulation/DTOs/UserWeatherRequest.cs
using WildfireSimulation.Models;

namespace WildfireSimulation.DTOs
{
    public class UserWeatherRequest
    {
        public LightningEvent? LightningEvent { get; set; }
        public bool? RandomLighting { get; set; }
        public WindEvent? WindEvent { get; set; }
        public bool? RandomWind { get; set; }
        public RainEvent? RainEvent { get; set; }
        public bool? RandomRain { get; set; }

    }
}
